core/domain/topic_services.py

```diff
@@ -325,7 +326,7 @@ def delete_skill(user_id, topic_id, skill_id):
     change_list = [topic_domain.TopicChange({
         'cmd': 'update_topic_property',
         'property_name': 'skill_ids',
-        'old_value': topic.skill_ids,
+        'old_value': copy.deepcopy(topic.skill_ids),
```

Comment:
Just a quick note to do this for any item at all in change list properties that's a list/dict as well. Might be worth double-checking the other models too for consistency/safety.

Alternatively you could do it in the TopicChange (and other) initializers.

Reply:
Shall I do this in TopicChange, StoryChange and SkillChange where self.new_value and old_value are equated with the one in the receiving dict?

Reply:
That seems reasonable to me.